src/components/maincontroller.ts

```diff
@@ -136,47 +131,71 @@ window.addEventListener('beforeunload', function () {
     reportPlaybackStopped($scope, getReportingParams($scope));
 });
 
-mgr.addEventListener(cast.framework.events.EventType.PLAY, (): void => {
-    play($scope);
-    reportPlaybackProgress($scope, getReportingParams($scope));
-});
+window.playerManager.addEventListener(
+    cast.framework.events.EventType.PLAY,
+    (): void => {
+        play($scope);
+    }
+);
 
-mgr.addEventListener(cast.framework.events.EventType.PAUSE, (): void => {
-    reportPlaybackProgress($scope, getReportingParams($scope));
-});
+window.playerManager.addEventListener(
+    cast.framework.events.EventType.PAUSE,
+    (): void => {
+        reportPlaybackProgress($scope, getReportingParams($scope));
+    }
+);
 
-function defaultOnStop(): void {
-    playbackMgr.stop();
+function defaultOnStopped(): void {
+    PlaybackManager.onStopped();
```

Comment:
```suggestion
function defaultOnStop(): void {
    PlaybackManager.onStop();
```

I don't understand very well this arbitrary name change

Reply:
I changed the names because we need to separate OnStopping from OnStopped.

The events are a mess and a good place to start is to know exactly when an event would fire. This new name is more explicit and allows us to know what will happen if we do certain things in here. The lack of this knowledge seems to have lead to a lot of weird behavior.

For example, this event would be used to report the playback location to the server. However, the playback already stopped so the position that got reported was at 0.